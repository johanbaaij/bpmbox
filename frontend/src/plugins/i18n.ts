import Vue from "vue";
import VueI18n, { LocaleMessages } from "vue-i18n";
import localesEn from "@/locales/en.json";

Vue.use(VueI18n);

const en: LocaleMessages = { en: localesEn };

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || "en",
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || "en",
  messages: en
});
