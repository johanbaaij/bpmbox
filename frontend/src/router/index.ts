import Vue from "vue";
import VueRouter from "vue-router";
import Home from "@/views/Home.vue";
import Register from "@/views/Register.vue";
import Login from "@/views/Login.vue";
import FourOhFour from "@/views/404.vue";
import Dashboard from "@/views/Dashboard.vue";
import EventBus from "@/utils/event-bus";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "home",
    component: Home
  },
  {
    path: "/404",
    name: "FourOhFour",
    component: FourOhFour
  },
  {
    path: "/register",
    name: "register",
    component: Register,
    meta: { auth: false }
  },
  {
    path: "/login",
    name: "login",
    component: Login,
    meta: { auth: false }
  },
  {
    path: "/dashboard",
    name: "dashboard",
    component: Dashboard,
    meta: { auth: true }
  }
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes
});

// add notification to "show on next page queue"
// before navigating, add notification to currently

router.beforeEach((_to, _from, next) => {
  next();
  EventBus.$emit("DISPLAY_NEXT_NOTIFICATION");
});

export default router;
