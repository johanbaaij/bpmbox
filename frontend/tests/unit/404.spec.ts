import "./setup";

// Dependencies
import Vuetify from "vuetify";
import i18n from "@/plugins/i18n";
import router from "@/router";

// Components
import FourOhFour from "../../src/views/404.vue";

// Utilities
import { mount, createLocalVue } from "@vue/test-utils";

const localVue = createLocalVue();

describe("FourOhFour.vue", () => {
  let vuetify: any;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it("should match snapshot", () => {
    const wrapper = mount(FourOhFour, {
      localVue,
      router,
      vuetify,
      i18n
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
