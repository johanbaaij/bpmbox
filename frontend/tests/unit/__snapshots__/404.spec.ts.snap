// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FourOhFour.vue should match snapshot 1`] = `
<div class="container container--fluid">
  <div class="row">
    <div class="col">
      <p>404</p>
      <p>
        Time to go <a href="/" class="router-link-exact-active router-link-active">home</a>.
      </p>
    </div>
  </div>
</div>
`;
